plugins/glustershd/glustershd.go

```diff
@@ -62,7 +60,13 @@ func (shd *Glustershd) Args() []string {
 
 // SocketFile returns path to the socket file used for IPC.
 func (shd *Glustershd) SocketFile() string {
-	return ""
+	if shd.socketfilepath != "" {
```

Comment:
Like mentioned before, this shouldn't be here.

The daemon framework will use `SocketFile()` to talk to the daemon. Here you are using `gdctx.MyUUID.String()` which is UUID of this node. Who is listening on this socket file ?